buildSrc/src/main/kotlin/java-library-conventions.gradle.kts

```diff
@@ -201,6 +204,23 @@ afterEvaluate {
 					"-parameters" // Generates metadata for reflection on method parameters.
 			))
 		}
+
+		val modulePath: String  = tasks.compileJava.get().classpath.asPath
+		val compileModuleInfoJava by tasks.registering(JavaCompile::class) {
+			dependsOn(tasks.compileJava)
+
+			destinationDir = tasks.compileJava.get().destinationDir
+			setSource("src/main/java/module-info.java")
+			classpath = files() // using module path below
+			options.encoding = "UTF-8"
+			options.compilerArgs.addAll(listOf(
+					"-Xlint",
+					"--release", "9",
+					"--module-version", "${project.version}",
+					"--module-path", modulePath
+			))
+		}
+		tasks.named("classes").get().dependsOn(compileModuleInfoJava)
```

Comment:
Is `tasks.named("classes")`  a `TaskProvider`? If so, I wonder if the call to `.get()` here could be removed as well?

Reply:
Oh sorry, I should have asked if you were ready to receive review comments yet @sormuras. :)

Reply:
hehe, your comments are welcome, @jbduncan -- will take them into account if we decide to use this approach, or when working with providers in general.